test/src/main/java/org/apache/accumulo/test/ComprehensiveTableOperationsIT.java

```diff
@@ -0,0 +1,957 @@
+    userTable = getUniqueNames(1)[0];
+
+    client.tableOperations().create(userTable);
+    ReadWriteIT.ingest(clientContext, 5, 5, 5, 0, userTable);
+    assertTrue(clientContext.getTabletLocationCache(TableId.of(ops.tableIdMap().get(userTable)))
+        .getTabletHostingRequestCount() > 0);
+    ops.clearLocatorCache(userTable);
+    assertEquals(0,
+        clientContext.getTabletLocationCache(TableId.of(ops.tableIdMap().get(userTable)))
+            .getTabletHostingRequestCount());
+
+    for (var sysTable : AccumuloTable.values()) {
+      assertEquals(0,
+          clientContext.getTabletLocationCache(sysTable.tableId()).getTabletHostingRequestCount());
+      ops.clearLocatorCache(sysTable.tableName());
+      assertEquals(0,
+          clientContext.getTabletLocationCache(sysTable.tableId()).getTabletHostingRequestCount());
+    }
+  }
+
+  @Test
+  public void
+      test_attachIterator_removeIterator_getIteratorSetting_listIterators_checkIteratorConflicts()
+          throws Exception {
+    // These may be tested elsewhere across several tests, but simpler to test all here
+    userTable = getUniqueNames(1)[0];
+    ops.create(userTable);
+    String iterName = "iter_name";
+    var iterSetting = new IteratorSetting(100, iterName, NoOpFilter.class);
+    var scope = EnumSet.of(IteratorUtil.IteratorScope.majc, IteratorUtil.IteratorScope.minc);
+    // system and user tables
+    Set<String> allTables = ops.list();
+
+    for (String table : allTables) {
+      ops.attachIterator(table, iterSetting, scope);
+      try {
+        assertTrue(ops.listIterators(table).containsKey(iterName));
+        assertNull(ops.getIteratorSetting(table, iterName, IteratorUtil.IteratorScope.scan));
+        assertEquals(iterSetting,
+            ops.getIteratorSetting(table, iterName, IteratorUtil.IteratorScope.majc));
+        assertThrows(AccumuloException.class,
+            () -> ops.checkIteratorConflicts(table, iterSetting, scope));
+        ops.checkIteratorConflicts(table, iterSetting, EnumSet.of(IteratorUtil.IteratorScope.scan));
+      } finally {
+        ops.removeIterator(table, iterName, scope);
+        assertFalse(ops.listIterators(table).containsKey(iterName));
+      }
+    }
+  }
+
+  @Test
+  public void test_addConstraint_listConstraints_removeConstraint() throws Exception {
+    // addConstraint, listConstraints, removeConstraint for user tables is tested in ConstraintIT.
+    // Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(ConstraintIT.class.getName()));
+    // addConstraint, listConstraints, removeConstraint not tested for system tables. Test basic
+    // functionality here
+    for (var sysTable : AccumuloTable.values()) {
+      var numExistingConstraints = ops.listConstraints(sysTable.tableName()).size();
+      String constraint = ComprehensiveBaseIT.TestConstraint.class.getName();
+
+      var constraintNum = ops.addConstraint(sysTable.tableName(), constraint);
```

Comment:
Should adding constraints on the system tables continue to be allowed or is this undesired?